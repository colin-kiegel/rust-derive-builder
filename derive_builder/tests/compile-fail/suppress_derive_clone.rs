#[macro_use]
extern crate derive_builder;

#[derive(Builder)]
#[builder(suppress_derive_clone)]
pub struct Example {
    field: String,
}

fn main() {
    let _ = ExampleBuilder::default().clone();
}
