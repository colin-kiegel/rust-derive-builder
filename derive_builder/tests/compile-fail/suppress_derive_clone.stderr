error[E0599]: no method named `clone` found for struct `ExampleBuilder` in the current scope
  --> tests/compile-fail/suppress_derive_clone.rs:11:39
   |
4  | #[derive(Builder)]
   |          ------- method `clone` not found for this
...
11 |     let _ = ExampleBuilder::default().clone();
   |                                       ^^^^^ method not found in `ExampleBuilder`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `clone`, perhaps you need to implement it:
           candidate #1: `Clone`
